### ISSUE TITLE
Nushell panic when trying to define module in terminal

### ISSUE THREAD

**Issue by WindSoilder** [nushell/nushell#6518]
### Describe the bug

As title, when trying to define module with `export def-env`, nushell panic

### How to reproduce

1. type the following:
```
module spam { export def-env
```
When we type the last character `v`, nushell will panic.

### Expected behavior

Nushell shouldn't panic

### Screenshots

[![asciicast](https://asciinema.org/a/J0DOQdOjCoq69d1oDAN83UbsG.svg)](https://asciinema.org/a/J0DOQdOjCoq69d1oDAN83UbsG)

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.68.0                                          |
| branch             |                                                 |
| commit_hash        | b0e5723a68d77b42d756879ab7773fdee81f8924        |
| build_os           | macos-x86_64                                    |
| build_target       | x86_64-apple-darwin                             |
| rust_version       | rustc 1.63.0 (4b91a6ea7 2022-08-08)             |
| rust_channel       | stable-x86_64-apple-darwin                      |
| cargo_version      | cargo 1.63.0 (fd9c4297c 2022-07-01)             |
| pkg_version        | 0.68.0                                          |
| build_time         | 2022-09-06 20:01:28 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by kubouch** [comment]
I'd say this is a different error. Can you compile in debug mode and provide the stack trace with `RUST_BACKTRACE=1`?

**Comment by WindSoilder** [comment]
Ok, here it is:
```
thread 'main' panicked at 'index out of bounds: the len is 2 but the index is 2', crates/nu-parser/src/parse_keywords.rs:898:63
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/panicking.rs:142:14
   2: core::panicking::panic_bounds_check
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/panicking.rs:84:5
   3: nu_parser::parse_keywords::parse_export_in_module
             at ./crates/nu-parser/src/parse_keywords.rs:898:63
   4: nu_parser::parse_keywords::parse_module_block::{{closure}}
             at ./crates/nu-parser/src/parse_keywords.rs:1393:56
   5: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/adapters/map.rs:84:28
   6: core::iter::traits::iterator::Iterator::fold
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/traits/iterator.rs:2366:21
   7: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/adapters/map.rs:124:9
   8: core::iter::traits::iterator::Iterator::for_each
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/traits/iterator.rs:783:9
   9: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/spec_extend.rs:40:17
  10: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/spec_from_iter_nested.rs:62:9
  11: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/spec_from_iter.rs:33:9
  12: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/mod.rs:2612:9
  13: core::iter::traits::iterator::Iterator::collect
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/traits/iterator.rs:1788:9
  14: <nu_protocol::ast::block::Block as core::convert::From<T>>::from
             at ./crates/nu-protocol/src/ast/block.rs:65:24
  15: <T as core::convert::Into<U>>::into
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/convert/mod.rs:550:9
  16: nu_parser::parse_keywords::parse_module_block
             at ./crates/nu-parser/src/parse_keywords.rs:1351:24
  17: nu_parser::parse_keywords::parse_module
             at ./crates/nu-parser/src/parse_keywords.rs:1499:13
  18: nu_parser::parser::parse_builtin_commands
             at ./crates/nu-parser/src/parser.rs:4811:22
  19: nu_parser::parser::parse_block::{{closure}}
             at ./crates/nu-parser/src/parser.rs:4990:43
  20: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/adapters/map.rs:84:28
  21: <core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits::iterator::Iterator>::fold::enumerate::{{closure}}
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/adapters/enumerate.rs:106:27
  22: core::iter::traits::iterator::Iterator::fold
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/traits/iterator.rs:2366:21
  23: <core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/adapters/enumerate.rs:112:9
  24: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/adapters/map.rs:124:9
  25: core::iter::traits::iterator::Iterator::for_each
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/traits/iterator.rs:783:9
  26: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/spec_extend.rs:40:17
  27: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/spec_from_iter_nested.rs:62:9
  28: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/spec_from_iter.rs:33:9
  29: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/alloc/src/vec/mod.rs:2612:9
  30: core::iter::traits::iterator::Iterator::collect
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/iter/traits/iterator.rs:1788:9
  31: <nu_protocol::ast::block::Block as core::convert::From<T>>::from
             at ./crates/nu-protocol/src/ast/block.rs:65:24
  32: <T as core::convert::Into<U>>::into
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/convert/mod.rs:550:9
  33: nu_parser::parser::parse_block
             at ./crates/nu-parser/src/parser.rs:4949:24
  34: nu_parser::parser::parse
             at ./crates/nu-parser/src/parser.rs:5401:9
  35: <nu_cli::syntax_highlight::NuHighlighter as reedline::highlighter::Highlighter>::highlight
             at ./crates/nu-cli/src/syntax_highlight.rs:20:30
  36: reedline::engine::Reedline::buffer_paint
             at /Users/chenhongze/.cargo/registry/src/github.com-1ecc6299db9ec823/reedline-0.11.0/src/engine.rs:1331:45
  37: reedline::engine::Reedline::repaint
             at /Users/chenhongze/.cargo/registry/src/github.com-1ecc6299db9ec823/reedline-0.11.0/src/engine.rs:1149:13
  38: reedline::engine::Reedline::read_line_helper
             at /Users/chenhongze/.cargo/registry/src/github.com-1ecc6299db9ec823/reedline-0.11.0/src/engine.rs:522:29
  39: reedline::engine::Reedline::read_line
             at /Users/chenhongze/.cargo/registry/src/github.com-1ecc6299db9ec823/reedline-0.11.0/src/engine.rs:411:22
  40: nu_cli::repl::evaluate_repl
             at ./crates/nu-cli/src/repl.rs:330:21
  41: nu::main
             at ./src/main.rs:366:31
  42: core::ops::function::FnOnce::call_once
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by merelymyself** [comment]
same problem for `export extern`.

**Comment by ehawman** [comment]
@hustcer Given that 0.68 was intended to be a graceful stepping stone from the old way of doing imports to the new method, yet the stepping stone turned out to have unexpected breaking changes, would this fix (and whatever comes from #6510) not be more suited for a smaller hotfix?

**Comment by hustcer** [comment]
@ehawman Yeah, there will be a patch version release
